docs/components/modal.md

```diff
@@ -1,348 +1,752 @@
+    </div>
+    <KInputSwitch
+      v-model="data.cancelEnabled"
+      label="Cancel button enabled"
+    />
+  </div>
 
-<div>
   <KModal
-    :isVisible="defaultIsOpen3"
-    title="Look Mah!"
-    content="I am customized!"
-    actionButtonText="Go"
-    actionButtonAppearance="creation"
-    cancelButtonText="Abort"
-    cancelButtonAppearance="danger"
-    @proceed="defaultIsOpen3 = false"
-    @canceled="defaultIsOpen3 = false"
+    :cancel-button-disabled="!data.cancelEnabled"
+    :visible="modal9Visible"
+    title="Modal"
+    @cancel="closeAllModals"
+    @proceed="closeAllModals"
   />
-</div>
+</KComponent>
 
 ```html
 <KModal
-  :isVisible="isVisible"
-  title="Look Mah!"
-  content="I am customized!"
-  actionButtonText="Go"
-  actionButtonAppearance="creation"
-  cancelButtonText="Abort"
-  cancelButtonAppearance="danger"
-  @proceed="isVisible = false"
-  @canceled="isVisible = false"
+  :cancel-button-disabled="false"
+  :visible="modalVisible"
+  title="Modal"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
 />
 ```
 
 ### hideCancelButton
 
-Use this to hide the built-in cancel button (`false` by default).
+Use this prop to hide the cancel button. Defaults to `false`.
 
-### hideDismissIcon
+<KButton @click="modal10Visible = true">Modal</KButton>
+<KModal
+  hide-cancel-button
+  :visible="modal10Visible"
+  title="Modal"
+  @cancel="closeAllModals"
+  @proceed="closeAllModals"
+/>
 
-When using the `header-image` slot we display a dismiss 'X' button in the upper right corner of the dialog. Set this prop to `true` to hide the button (`false` by default).
+```html
+<KModal
+  hide-cancel-button
+  :visible="modalVisible"
+  title="Modal"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
+/>
+```
 
-::: tip
-If you want to have a dismiss icon on your dialog without using the `header-image` slot, you should use the [KPrompt](/components/prompt) component.
-:::
+### hideCloseIcon
 
-### dismissButtonTheme
+Prop that allows you to hide close icon next to the title. Defaults to `false`. When no title is passed and `hideCloseIcon` is `true`, the modal header section will be hidden.
 
-Can be `dark` (default) or `light`. You might want to use this if displaying dark content in the `header-image` slot.
+<KButton @click="modal11Visible = true">Modal</KButton>
+<KModal
+  hide-close-icon
+  :visible="modal11Visible"
+  @cancel="closeAllModals"
+  @proceed="closeAllModals"
+>
+  Modal header section is omitted when no header content is present.
+</KModal>
 
-<KButton appearance="primary" @click="slottedIsOpen2 = true">Open Onboarding Modal</KButton>
+```html
+<KModal
+  hide-close-icon
+  :visible="modalVisible"
+  @cancel="handleModalClose"
+  @proceed="handleModalProceed"
+>
+  Modal header section is omitted when no header content is present.
+</KModal>
+```
+
+### proceedOnEnter
+
+Whether pressing on Enter should result in KModal emitting the [`proceed` event](#proceed). Defaults to `true`.
```

Comment:
This example does not function properly on the docs site; modal stays open and nothing happens.

Reply:
I think it's because the element isn't focused when opened - this is different behavior than existed previously, so it's broken

Reply:
I'm not sure this makes sense -- hitting "Enter" on the modal/prompt doesn't do anything regardless. Did this break prior to all of this alpha work?

@kaiarrowood I think you originally implemented this?

Reply:
It's working - you gotta blur from the KButton: https://www.loom.com/share/d8bc979195c342fb9ce50011b0a751a4?sid=4bd23e94-0cd9-4319-9c25-f2e8ebbba403

Reply:
This didn't used to be the case. IIRC this prop was added because at one point when the dialog opened you could immediately press enter; otherwise, the prop doesn't make sense..?

Reply:
I think it works as expected. The focus is set on the button, not on KModal. When I don't set focus on anything - it closes on Enter: https://www.loom.com/share/8c340b2c79524eccaf5c28cb40165fa0?sid=8ce509ca-61dc-4b98-b0e0-e60e7b017feb

Reply:
What I'm saying is that the modal/prompt should be automatically focused, or an element within it, when opened. This used to work this way, so there has been a regression at some point.

Reply:
Fixed now (both in sandbox and in docs)

Reply:
Chiming in on expected behavior. As far as focus trap goes, I don't think we ever had it working that `Enter` would immediately do something (I think that's an accessibility no-no, because we would manually be grabbing the user's focus and dropping it wherever we chose which could be confusing for non-sighted) - the only place I did see this behavior was v7 KModalFullscreen (didn't check v8). 

KPrompt/KModal have never supported immediate close with `Enter` and honestly not sure they should. We SHOULD allow immediate close with `Esc` though, that doesn't require any focus change - it's just treated as a native keyboard action that always tries to exit out of whatever is focused.

Reply:
So then what is the purpose of the `preventProceedOnEnter` prop? How did it ever function?

Reply:
`preventProceedOnEnter` is breaking the tie between pressing Enter key and sending the proceed event when focused on the dialog. I believe the thinking was something like, if the user completely controls the buttons by using the footer slot, we don't want to send a proceed event since we don't know the state of the buttons (ie. you wouldn't send `@proceed` if the button is disabled).

Reply:
...then the prop can also be removed

Reply:
I'm fine with this approach. If needed, host app can take care of performing an action on enter:
```
// inside modal body
<KInput @enter.prevent="...
```

Reply:
Removed the prop